tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -844,6 +846,27 @@ ROOT::Experimental::Internal::RNTupleDescriptorBuilder::AddColumn(RColumnDescrip
    return RResult<void>::Success();
 }
 
+ROOT::Experimental::RResult<void>
+ROOT::Experimental::Internal::RNTupleDescriptorBuilder::AttachColumn(DescriptorId_t fieldId,
+                                                                     DescriptorId_t logicalColumnId)
+{
+   auto itrFieldDesc = fDescriptor.fFieldDescriptors.find(fieldId);
+   if (itrFieldDesc == fDescriptor.fFieldDescriptors.end()) {
+      return R__FAIL("AttachColumn: invalid field ID");
+   }
+   const auto itrColumnDesc = fDescriptor.fColumnDescriptors.find(logicalColumnId);
+   if (itrColumnDesc == fDescriptor.fColumnDescriptors.cend()) {
+      return R__FAIL("AttachColumn: invalid column ID");
+   }
+   const auto columnIdx = itrColumnDesc->second.GetIndex();
+   if (itrFieldDesc->second.fLogicalColumnIds.size() <= columnIdx) {
+      itrFieldDesc->second.fLogicalColumnIds.resize(columnIdx + 1);
```

Comment:
Do we need to initialize the intermediate elements with `kInvalidDescriptorId`?
```c++
itrFieldDesc->second.fLogicalColumnIds.resize(columnIdx + 1, kInvalidDescriptorId);
```

Reply:
(ah, I see you do this in the third commit; please consider squashing its changes into the first commit, we should not have potentially problematic code in the history)